Do plan to add a test to create an instance of the class, which will either return no credentials or (in EC2), actually work. 

Tests run against s3 express

one failure for @ahmarsuhail to worry about.
```
[ERROR]   ITestS3AAnalyticsAcceleratorStreamReading.testSequentialStreamsNoDuplicateGets:402 [Counter named action_http_get_request with expected value 1] 
Expecting:                                                                                                                                                                                          
 <2L>                                                                                                                                                                                               
to be equal to:                                                                                                                                                                                     
 <1L>  
```


and two failures in assume roles of malformed roles. Looks like STS has changed its error text. fix: remove the probes for specific text
```
  ITestAssumeRole.testAssumeRoleFSBadPolicy:251->expectFileSystemCreateFailure:164  Expected to find 'JSON' but got unexpected exception: org.apache.hadoop.fs.s3a.AWSBadRequestException: Instantiate org.apache.hadoop.fs.s3a.auth.AssumedRoleCredentialProvider on /: software.amazon.awssdk.services.sts.model.MalformedPolicyDocumentException: Unexpected IOException: Unexpected close marker '}': expected ']' (for ROOT starting at [Source: java.io.StringReader@16a63c4d; line: 1, column: 0])                                                                                           
 at [Source: java.io.StringReader@16a63c4d; line: 1, column: 2] (Service: Sts, Status Code: 400, Request ID: 47f1da5d-c400-4269-866a-7324cb167a1a) (SDK Attempt Count: 1):MalformedPolicyDocument: Unexpected IOException: Unexpected close marker '}': expected ']' (for ROOT starting at [Source: java.io.StringReader@16a63c4d; line: 1, column: 0])                                                 
 at [Source: java.io.StringReader@16a63c4d; line: 1, column: 2] (Service: Sts, Status Code: 400, Request ID: 47f1da5d-c400-4269-866a-7324cb167a1a) (SDK Attempt Count: 1)                           
        at org.apache.hadoop.fs.s3a.S3AUtils.translateException(S3AUtils.java:271)                                                                                                                  
        at org.apache.hadoop.fs.s3a.S3AUtils.getInstanceFromReflection(S3AUtils.java:705)                                                                                                           
        at org.apache.hadoop.fs.s3a.auth.CredentialProviderListFactory.createAWSV2CredentialProvider(CredentialProviderListFactory.java:303)                                                        
        at org.apache.hadoop.fs.s3a.auth.CredentialProviderListFactory.buildAWSProviderList(CredentialProviderListFactory.java:249)                                                                 
        at org.apache.hadoop.fs.s3a.auth.CredentialProviderListFactory.createAWSCredentialProviderList(CredentialProviderListFactory.java:142)                                                      
        at org.apache.hadoop.fs.s3a.S3AFileSystem.createClientManager(S3AFileSystem.java:1151)                                                                                                      
        at org.apache.hadoop.fs.s3a.S3AFileSystem.initialize(S3AFileSystem.java:729)                                                                                                                
        at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:3616)                                                                                                                   
        at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:555)                                                                                                                                 
        at org.apache.hadoop.fs.Path.getFileSystem(Path.java:373)                                                                                                                                   
        at org.apache.hadoop.fs.s3a.auth.ITestAssumeRole.lambda$expectFileSystemCreateFailure$0(ITestAssumeRole.java:166)                                                                           
        at org.apache.hadoop.fs.s3a.S3ATestUtils.lambda$interceptClosing$0(S3ATestUtils.java:753)                       
```

And here

```
  ITestAssumeRole.testAssumeRoleFSBadPolicy2:262->expectFileSystemCreateFailure:164  Expected to find 'Syntax errors in policy' but got unexpected exception: org.apache.hadoop.fs.s3a.AWSBadRequestException: Instantiate org.apache.hadoop.fs.s3a.auth.AssumedRoleCredentialProvider on /: software.amazon.awssdk.services.sts.model.MalformedPolicyDocumentException: Unexpected IOException: Unexpected character (''' (code 39)): was expecting double-quote to start field name                                                                                                               
 at [Source: java.io.StringReader@21d2e81f; line: 1, column: 3] (Service: Sts, Status Code: 400, Request ID: 7b22250d-1238-4609-92f2-280531aacf11) (SDK Attempt Count: 1):MalformedPolicyDocument: Unexpected IOException: Unexpected character (''' (code 39)): was expecting double-quote to start field name                                                                                         
 at [Source: java.io.StringReader@21d2e81f; line: 1, column: 3] (Service: Sts, Status Code: 400, Request ID: 7b22250d-1238-4609-92f2-280531aacf11) (SDK Attempt Count: 1) 
```

this is all really good for production use: callers are getting errors back from the parser (notable that this change coincides with re-invent). But our tests fail...

--- a/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/auth/IAMInstanceCredentialsProvider.java
+++ b/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/auth/IAMInstanceCredentialsProvider.java
@@ -67,7 +67,7 @@ public class IAMInstanceCredentialsProvider
    * How far in advance of credential expiry must IAM credentials be refreshed.
    * See HADOOP-19181. S3A: IAMCredentialsProvider throttling results in AWS auth failures
    */
-  public static final Duration TIME_BEFORE_EXPIRY = Duration.ofMinutes(1);
+  public static final Duration TIME_BEFORE_EXPIRY = Duration.ofMinutes(5);
 
   /**
    * The credentials provider.

--- a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/TestS3AAWSCredentialsProvider.java
+++ b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/TestS3AAWSCredentialsProvider.java
@@ -55,11 +55,13 @@ import org.apache.hadoop.fs.s3a.auth.AssumedRoleCredentialProvider;
 import org.apache.hadoop.fs.s3a.auth.CredentialProviderListFactory;
 import org.apache.hadoop.fs.s3a.auth.IAMInstanceCredentialsProvider;
 import org.apache.hadoop.fs.s3a.auth.NoAuthWithAWSException;
+import org.apache.hadoop.fs.s3a.auth.NoAwsCredentialsException;
 import org.apache.hadoop.fs.s3a.auth.ProfileAWSCredentialsProvider;
 import org.apache.hadoop.fs.s3a.auth.delegation.CountInvocationsProvider;
 import org.apache.hadoop.fs.s3a.impl.InstantiationIOException;
 import org.apache.hadoop.fs.s3a.test.PublicDatasetTestUtils;
 import org.apache.hadoop.io.retry.RetryPolicy;
+import org.apache.hadoop.test.HadoopTestBase;
 import org.apache.hadoop.util.Sets;
 
 import static org.apache.hadoop.fs.s3a.Constants.ASSUMED_ROLE_CREDENTIALS_PROVIDER;
@@ -79,7 +81,7 @@ import static org.apache.hadoop.util.StringUtils.STRING_COLLECTION_SPLIT_EQUALS_
 /**
  * Unit tests for {@link Constants#AWS_CREDENTIALS_PROVIDER} logic.
  */
-public class TestS3AAWSCredentialsProvider extends AbstractS3ATestBase {
+public class TestS3AAWSCredentialsProvider extends HadoopTestBase {
 
   /**
    * URI of the test file: this must be anonymously accessible.
@@ -921,4 +923,19 @@ public class TestS3AAWSCredentialsProvider extends AbstractS3ATestBase {
     }
   }
 
+  @Test
+  public void testIAMInstanceCredentialsProvider() throws Throwable {
+
+    final Configuration conf =
+        createProviderConfiguration(IAMInstanceCredentialsProvider.class);
+    Path testFile = getExternalData(conf);
+    try (AWSCredentialProviderList list = createAWSCredentialProviderList(testFile.toUri(), conf)) {
+      try {
+        list.resolveCredentials();
+      } catch (NoAwsCredentialsException e) {
+        ///  this is OK.
+      }
+
+    }
+  }
 }

--- a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/auth/ITestAssumeRole.java
+++ b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/auth/ITestAssumeRole.java
@@ -102,6 +102,8 @@ public class ITestAssumeRole extends AbstractS3ATestBase {
   private static final Statement STATEMENT_ALL_BUCKET_READ_ACCESS
       = statement(true, S3_ALL_BUCKETS, S3_BUCKET_READ_OPERATIONS);
 
+  public static final String MALFORMED_POLICY_DOCUMENT = "MalformedPolicyDocument";
+
   /**
    * test URI, built in setup.
    */
@@ -244,13 +246,13 @@ public class ITestAssumeRole extends AbstractS3ATestBase {
 
   @Test
   public void testAssumeRoleFSBadPolicy() throws Exception {
-    describe("Attemnpt to create the FS with malformed JSON");
+    describe("Attempt to create the FS with malformed JSON");
     Configuration conf = createAssumedRoleConfig();
     // add some malformed JSON
     conf.set(ASSUMED_ROLE_POLICY,  "}");
     expectFileSystemCreateFailure(conf,
         AWSBadRequestException.class,
-        "JSON");
+        MALFORMED_POLICY_DOCUMENT);
   }
 
   @Test
@@ -261,7 +263,7 @@ public class ITestAssumeRole extends AbstractS3ATestBase {
     conf.set(ASSUMED_ROLE_POLICY, "{'json':'but not what AWS wants}");
     expectFileSystemCreateFailure(conf,
         AWSBadRequestException.class,
-        "Syntax errors in policy");
+        MALFORMED_POLICY_DOCUMENT);
   }
 
   @Test

